app/src/main/java/org/mozilla/fenix/browser/BrowserStore.kt

```diff
@@ -0,0 +1,26 @@
+package org.mozilla.fenix.browser
+
+import mozilla.components.browser.state.action.BrowserAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.feature.media.ext.findActiveMediaTab
+import mozilla.components.lib.state.Store
+
+fun BrowserStore.waitForPIPToClose(
```

Comment:
Yes.. Actually after going through waitForSelectedOrDefaultSearchEngine in BrowserStore, I thought that we can implement similar solution to avoid the race. Will move extension to fenix/ext and also add tests.